cdf-fabric-replicator-chart/templates/deployment.yaml

```diff
@@ -0,0 +1,152 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "cdf-fabric-replicator-chart.fullname" . }}
+  labels:
+    {{- include "cdf-fabric-replicator-chart.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "cdf-fabric-replicator-chart.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "cdf-fabric-replicator-chart.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: COGNITE_TOKEN_URL
+              valueFrom:
+                configMapKeyRef:
+                  name: cdf-conn-config
+                  key: COGNITE_TOKEN_URL
+            - name: COGNITE_CLIENT_ID
+              valueFrom:
+                configMapKeyRef:
+                  name: cdf-conn-config
+                  key: COGNITE_CLIENT_ID
+            - name: COGNITE_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
```

Comment:
When I last used `secretKeyRef`, the way I used it was that I created a secret using `kubectl create secret` on the kubernetes cluster then was able to reference a key in the secret in the helm chart.  Is that how this works here?

Reply:
That is one way you can initially inject the secret we create the secret in the secret yaml file and that is seeded by the values.yaml